common/kvstore/src/main/java/org/apache/spark/util/kvstore/RocksDBIterator.java

```diff
@@ -270,20 +280,30 @@ static int compare(byte[] a, byte[] b) {
     return a.length - b.length;
   }
 
-  private record ResourceCleaner(RocksIterator rocksIterator, RocksDB rocksDB) implements Runnable {
+  private static class ResourceCleaner implements Runnable {
+
+    private final RocksIterator rocksIterator;
+    private final RocksDB rocksDB;
+    private final AtomicBoolean status = new AtomicBoolean(true);
+
+    public ResourceCleaner(RocksIterator rocksIterator, RocksDB rocksDB) {
+      this.rocksIterator = rocksIterator;
+      this.rocksDB = rocksDB;
+    }
 
     @Override
     public void run() {
-      rocksDB.getIteratorTracker().removeIf(ref -> {
-        RocksDBIterator<?> rocksDBIterator = ref.get();
-        return rocksDBIterator != null && rocksIterator.equals(rocksDBIterator.it);
-      });
-      synchronized (rocksDB.getRocksDB()) {
-        org.rocksdb.RocksDB _db = rocksDB.getRocksDB().get();
-        if (_db != null) {
-          rocksIterator.close();
-        }
+      if (status.compareAndSet(true, false)) {
+        rocksDB.getIteratorTracker().removeIf(ref -> {
```

Comment:
Is it possible to reuse the `db.notifyIteratorClosed`?

Reply:
Thanks, updated, levels will update after rocksdb complete